app/components/contacts/AllContacts.js

```diff
@@ -57,8 +70,10 @@ class AllContacts extends Component<Props, State> {
   };
 
   render() {
-    const { contacts, lastUsedAddresses, addContact, selectContact, isModalMode } = this.props;
+    const { contacts, transactions, addContact, selectContact, isModalMode } = this.props;
     const { searchTerm } = this.state;
+    const lastUsedAddresses: TxList = getFirstUniqueTransactions(transactions);
```

Comment:
there is another component where you make same calculation. Maybe there is a way to save and update last used addresses in single place without any heavy lifting and shortcuts? _Reducer?_

Reply:
moved logic to reducer